extension/chrome/elements/compose-modules/compose-input-module.ts

```diff
@@ -78,11 +78,20 @@ export class ComposeInputModule extends ViewModule<ComposeView> {
     return this.view.sendBtnModule.popover.choices.richtext;
   };
 
+  public checkInputLengthBeforePasting = (textToPaste: string, targetInputField: HTMLElement, ev: ClipboardEvent) => {
+    const currentLength = targetInputField.innerText.length;
+    const limit = 50000;
+    if (textToPaste.length + currentLength > limit) {
+      ev.preventDefault();
+    }
+  };
+
   private handlePaste = () => {
     this.squire.addEventListener('willPaste', (e: WillPasteEvent) => {
       const div = document.createElement('div');
       div.appendChild(e.fragment);
       const html = div.innerHTML;
+      this.checkInputLengthBeforePasting(div.innerText, this.view.S.cached('input_text').get(0), e);
```

Comment:
Instead of copying `this.view.S.cached('input_text').get(0)` it'd be more clean to extract the element from the `squire` object itself. `SquireEditor` actually has a method `getRoot()` for this. It was missing in our `squire.d.ts` type definitions, so I added it. You can now use `this.squire.getRoot()`

Another remark: is there a good reason to use `div.innerText` for the check because what actually used is sanitized `div.innerHTML`. Can it be that we inserting a big load of tags with little text? Can we perform the check on the `sanitized` value instead?

Also, when you `preventDefault`, I guess, further manipulations on `e.fragment` serve no purpose? If so, they should be skipped. It'd be more convenient to have the check method to return a `boolean` value and then we can either `ev.preventDefault()` or do the rest steps. The method may be called `willInputLimitBeExceeded`

Reply:
Thank you for all the remarks. I apply some changes according to your helpful insight.